features/plugin.feature

```diff
@@ -547,3 +548,30 @@ Feature: Manage WordPress plugins
     Then STDOUT should be a table containing rows:
       | name         | title | description                    |
       | db-error.php |       | Custom database error message. |
+
+  @require-wp-4.0
+  Scenario: Validate installed plugin's version.
+    Given a WP installation
+    And I run `wp plugin install hello-dolly`
+    And a wp-content/mu-plugins/test-plugin-update.php file:
+      """
+      <?php
+      /**
+       * Plugin Name: Test Plugin Update
+       * Description: Fakes installed plugin's data to verify plugin version mismatch
+       * Author: WP-CLI tests
+       */
+
+      add_filter( 'site_transient_update_plugins', function( $value ) {
+          if ( isset( $value->no_update['hello-dolly/hello.php'] ) ) {
```

Comment:
I think this could potentially cause sporadic test failures where an update will be noticed after the installation step in line 555.

I think it would be better to do the following instead, without a prior check:
- `unset ( $value->response['hello-dolly/hello.php'] );` (in case an actual update was found)
- `$value->no_update['hello-dolly/hello.php']->new_version = '1.5';` to fake the versions as you already did.

This should always reliable trigger the error condition, if I've read the code correctly.